include/boost/math/ccmath/detail/swap.hpp

```diff
@@ -0,0 +1,21 @@
+//  (C) Copyright Matt Borland 2021.
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_MATH_CCMATH_DETAIL_SWAP_HPP
+#define BOOST_MATH_CCMATH_DETAIL_SWAP_HPP
+
+namespace boost::math::ccmath::detail {
+
+template <typename T>
+inline constexpr void swap(T& x, T& y) noexcept
+{
+    T temp = x;
+    x = y;
+    y = temp;
+}
```

Comment:
@mborland : Is there no constexpr `std::swap`?

Reply:
`std::swap` only became `constexpr` in [C++20.](https://en.cppreference.com/w/cpp/algorithm/swap)